compiler/src/codegen/transl_anf.re

```diff
@@ -626,6 +627,12 @@ let rec compile_comp = (env, c) => {
     switch (c.comp_desc) {
     | CSwitch(arg, branches) =>
       let compiled_arg = compile_imm(env, arg);
+      let switch_type =
```

Comment:
Can a switch be empty? If not, we should maybe have something that raises an error here rather than silently defaults to `I32Type`, since the symptom of us breaking that invariant in the future will be bizarre type errors.

Reply:
It's less that the switch is empty and more that we don't know what the type is, for example:
```
let foo = (a) => {
  match (5) {
    _ => a
  }
}
```

Grain values are always `I32Type`, so it's a sane default. You're not allowed to use any types when working with the unsafe wasm values, so this is OK. (I was going to say that was called out in the document I wrote up about it but I think I never wrote it 🙃. I'll get on that.)